lib/iris/tests/test_load.py

```diff
@@ -11,17 +11,41 @@
 # import iris tests first so that some things can be initialised before importing anything else
 import iris.tests as tests  # isort:skip
 
+import pathlib
+
 import iris
 import iris.io
 
 
+@tests.skip_data
+class TestGenerateCubes(tests.IrisTest):
```

Comment:
In this file we're integration testing load and its variants - all the way down to making the cube(s). For `_generate_cubes` I think it would be better to do something tighter wrapped around it so that we only get a failure in the test if `_generate_cubes` itself is to blame.

To do this, I suggest a file similar to `lib\iris\tests\unit\io\test_expand_filespecs.py` that mocks `decode_uri` and checks for it being passed the correct thing. I'm not quite sure how that should look but I didn't want to delay getting the rest of this review out so I'll get back to you on that.

Reply:
My suggestion would be a test function that does something like what's below. I'm happy to write that if you'd like, rather than put plumbing this all on you, but no pressure either way.

```
test_variants = [
    ("string", "string"),
    (["string"], "string"),
    (Path("string"), Path("string")),
]

decode_uri_mock = self.patch("iris.iris.io.decode_uri", return_value=('file', None))
self.patch("iris.iris.io.load_files")

for gc_arg, du_arg in test_variants:
    decode_uri_mock.reset_mock()
    list(iris._generate_cubes(gc_arg, None, None))
    decode_uri_mock.assert_called_with(du_arg)
```

Reply:
I've removed this change.  Would you be happy to create the new file for this test?

Reply:
I added a file next to it for a save unit test too (and populated them)